runtime-modules/project-token/src/types.rs

```diff
@@ -370,12 +375,16 @@ impl<AccountId, Balance> From<Transfers<AccountId, Balance>>
 /// Block Rate bare minimum impementation
 impl BlockRate {
     pub fn from_yearly_rate(r: YearlyRate, blocks_per_year: u32) -> Self {
-        BlockRate(Permill::from_parts(blocks_per_year).saturating_mul(r.0))
+        BlockRate(Permill::from_rational_approximation(
+            blocks_per_year,
+            r.0.saturating_reciprocal_mul(blocks_per_year),
+        ))
```

Comment:
First of all
```
Permill::from_rational_approximation(
            blocks_per_year,
            r.0.saturating_reciprocal_mul(blocks_per_year),
        )
```
Will just be `r.0`.
This can be checked with a simple unit test:
```
#[test]
fn block_conversion_unit_test() {
    let one_percent_per_year = yearly_rate!(1);
    assert_eq!(
        BlockRate::from_yearly_rate(one_percent_per_year, 100).0,
        // We expect 0,01%, so 100 / 1 000 000 (100 per mill)
        Permill::from_parts(100)
    )
}
```
Which will result in:
```
---- tests::patronage::block_conversion_unit_test stdout ----
thread 'tests::patronage::block_conversion_unit_test' panicked at 'assertion failed: `(left == right)`
  left: `Permill(10000)`,
 right: `Permill(100)`', runtime-modules/project-token/src/tests/patronage.rs:16:5
```

Another thing to note is that **you cannot use `Permill` to represent both `YearlyRate` and `BlockRate`.**
The block rate should be `1 / 5256000` of the yearly rate, so even if the yearly rate is 100%, the block rate will be so small that `Permill` will be insufficient to represent it (effectively `1 / 5256000` would be just floored to `0`).

You need a `Perquintill` block rate to be able to represent a yearly rate which is `Permill`.

Reply:
The min annual percentage is `1% = .01` divided by the blocks per year = `5256000`  is `1.90128527e-9`
that is `1901285270 [1/ (billion * billion)] = 1901285270 [1/ quintillion]` (https://en.wikipedia.org/wiki/Names_of_large_numbers)
This means if I set `YearlyRate = Percent` and `BlockRate = Perquintill`

Reply:
- `1% [1/ year]` is `.01 [1/year]` 
- `.01 [1/year] / 5256000 [blocks / year]` is `1.90128527e-9 [1 / blocks]`
`1.90128527e-9` is `1901285270 [1/quintillions]` (parts per quintillion)
-> I used `YearlyRate = Percent` and `BlockRate = Perquintill`

Reply:
fixed here: 6e1e2c8137f7c46425102d01c0c54424dd9e4573, the commit has an explainer body